jax/_src/api.py

```diff
@@ -321,27 +343,18 @@ def _python_jit(
   def f_jitted(*args, **kwargs):
     if config.jax_disable_jit:
       return fun(*args, **kwargs)
-    if max(donate_argnums, default=-1) >= len(args):
-      raise ValueError(
-          f"jitted function has donate_argnums={donate_argnums} but "
-          f"was called with only {len(args)} positional arguments.")
-
-    f = lu.wrap_init(fun)
-    f, args = argnums_partial_except(f, static_argnums, args,
-                                     allow_invalid=True)
-    f, kwargs = argnames_partial_except(f, static_argnames, kwargs)
-    args_flat, in_tree = tree_flatten((args, kwargs))
-    if donate_argnums:
-      donated_invars = donation_vector(donate_argnums, args, kwargs)
-    else:
-      donated_invars = (False,) * len(args_flat)
-    for arg in args_flat:
-      _check_arg(arg)
-    flat_fun, out_tree = flatten_fun(f, in_tree)
-    out = xla.xla_call(flat_fun, *args_flat, device=device, backend=backend,
-        name=flat_fun.__name__, donated_invars=donated_invars, inline=inline)
-    return tree_unflatten(out_tree(), out)
+    closed_fun, in_tree, args_flat, donated_invars = _prepare_jit(
```

Comment:
consider returning a dict you later splat as kwargs? that's what I do with `xmap` and `pjit` to avoid messing up the order of returns/binders

Reply:
I also called it `_infer_params`, which I think is a better description for this function than `_prepare_jit`, but I don't have a super strong preference